src/vfs2.c

```diff
@@ -1505,7 +1505,9 @@ int vfs2_poll(sqlite3_file *file,
 		sl->len = len;
```

Comment:
We are setting len on 1502 already.

Reply:
Also, I cannot comment on the line but the code above from lines 1493 to 1497 looks a little bit weird because:
* if we dont care about the **frames and *n then it frees `pending_txn_frames`,
* but if we care about **frames and *n then they get populated based on pending_txt_frames and we keep the latter alive because we are assuming something about the usage of this function from the outside if that makes sense.

For the second case, wouldn't it be simpler if we returned the pointer to the frames and cleared `pending_txn`?

Reply:
In the second case we don't keep `pending_txn_frames` alive, we free them---what we do is to avoid clearing the value of `pending_txn_start` and `pending_txn_len` so that `vfs2_unhide` can do its work later. I think this could all be written more clearly.

Reply:
I've pushed a commit that refactors this (eff99ca), let me know if it's an improvement!